### PR TITLE
Fix conversion rate for Fixer and Unicredit provider, #28279

### PR DESCRIPTION
**Pull request by Joozty**
https://is.roihunter.com/issues/28279

There was a problem with a calculation to base currency. See commits. :)

I am not sure about Unicredit provider. It sends wrong rates IMHO. For example for 2019-03-27 it returned 1 CZK = 8.153 HUF. Which is super off. The current value is 1 CZK = 12.50 HUF

I am not sure what about wrong values in DB. I think we should delete it.

```
DELETE FROM "USD_exchange_rates" WHERE date='2019-03-11' AND provider_id in (5,6)
```

### CODE REVIEW

File: gold_digger/data_providers/_provider.py [hunk 1]
```diff
@@ -104,8 +104,8 @@ def _conversion_to_base_currency(self, base_currency_rate, currency_rate, logger
         :type logger: gold_digger.utils.ContextLogger
         :rtype: decimal.Decimal
         """
-        conversion = 1 / currency_rate
-        return self._to_decimal(base_currency_rate * conversion, logger=logger)
+        conversion = 1 / base_currency_rate
+        return self._to_decimal(currency_rate * conversion, logger=logger)
```
Comment (harpener) [inline gold_digger/data_providers/_provider.py]: @Joozty We were discussing this with @DavidPrexta and agreed on that it was wrong before. Are you saying we concluded wrong? :D 😞

Comment (Joozty) [reply gold_digger/data_providers/_provider.py]: Well, I am not good at math but I guess you were wrong. :smile:

File: tests/data_providers/test_unicredit.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+
+from decimal import Decimal
+
+
+def test_fixer_conversion_to_base_currency(unicredit, logger):
```
Comment (harpener) [inline tests/data_providers/test_unicredit.py]: @Joozty You probably want to rename the test. :-)

File: tests/data_providers/test_unicredit.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    """
+    Unicredit has CZK as a base currency. It has to be converted firstly to USD.
+
+    Example Fixer response:
```
Comment (harpener) [inline tests/data_providers/test_unicredit.py]: @Joozty And also this should be Unicredit. :-)

Comment (Joozty) [reply tests/data_providers/test_unicredit.py]: https://github.com/business-factory/gold-digger/pull/62/commits/b251376302ef3f102b9c8e146dde6ef717c0792f

File: tests/data_providers/test_unicredit.py (hunk 2 above)
Comment (Joozty) [reply tests/data_providers/test_unicredit.py]: https://github.com/business-factory/gold-digger/pull/62/commits/b251376302ef3f102b9c8e146dde6ef717c0792f

**Comment by harpener** [comment]
Fixer will return two exchange rates for each of two symbols of currencies (e.g. USD, HUF), having EUR as base.
The rates will mean how much USD is 1 EUR and how much HUF is 1 EUR.

1 EUR = 319.9 HUF
1 EUR = 1.13 USD

1.13 USD = 319.9 HUF
USD = (319.9 / 1.13) HUF = 275 HUF


Unicredit will return all exchange rates for all supported currencies (except CZK), having CZK as base.
The rates will mean how much CZK is 1 USD and how much CZK is 1 HUF.

1 USD = 22 CZK
1 HUF = 0.08 CZK

1 CZK = 1/22 USD
1 CZK = 1/0.08 HUF

1/22 USD = 1/0.08 HUF
USD = (1/0.08 / 1/22) HUF = 275 HUF

The problem is that HUF returns 8 instead of 0.08. But it does return 22 for USD. Maybe remove the Unicredit after all, because it seems messy to me. :-)

**Comment by Joozty** [comment]
@harpener Thanks check my latest two commits. :)

**Comment by harpener** [comment]
@business-factory/pythonistas I deleted unicredit provider and all rates for unicredit and fixer providers from master database.

File: gold_digger/data_providers/_provider.py (hunk 1 above)
Comment (martinvy) [reply gold_digger/data_providers/_provider.py]: @harpener So, you were wrong when you changed the formula in #61, yes? Well, at least I am glad I implemented it correctly in the beginning 😛

**Pull request merged**